src/odemis/driver/xt_client.py

```diff
@@ -1990,3 +2002,178 @@ def stop(self, axes=None):
             self._updatePosition()
         except Exception:
             logging.exception("Unexpected failure when updating position")
+
+
+class MultiBeamScanner(Scanner):
+    """
+    This class extends  behaviour of the xt_client.Scanner class with XTtoolkit functionality.
+    xt_client.Scanner contains Vigilant Attributes for magnification, accel voltage, blanking, spotsize, beam shift,
+    rotation and dwell time. This class adds XTtoolkit functionality via the Vigilant Attributes for the pitch,
+    beam stigmator, pattern stigmator, the beam shift transformation matrix (read-only),
+    multiprobe rotation (read-only), aperture index, beamlet index, beam mode (multi/single beam)
+    Whenever one of these attributes is changed, its setter also updates another value if needed.
+    """
+
+    def __init__(self, name, role, parent, hfw_nomag, **kwargs):
+        # First instantiate the xt_toolkit VA's then call the __init__ of the super for the update thread.
+        self.parent = parent
+
```

Comment:
I wonder if it makes sense to have a check if XTToolkit is running in this constructor. Right now if the type on the adapter side is set to xtlib and you try to start a MultiBeamScanner it will fail because of a NotImplementedError when calling `self.parent.pitch_info()` on line 2022. 
It might be useful to have a check for XTToolkit and give a descriptive error message. What do you think?

Reply:
Would you do that simply via:

```
        if "xttoolkit" in self.microscope.swVersion.lower():
            self.xt_type = "xttoolkit"
        else:
            raise TypeError("Xttoolkit must be running for this test.")
```
            
 And then implement it in the scanner of SEM constructor? (SEM part in the if statement where we instatiate the MB-scanner part)

Reply:
I think the SEM constructor makes the most sense. Then it fails the soonest, which I guess saves time. In the SEM constructor you will need `self._swVersion`. And don't forget to update the text in the raised error.

```
        if "xttoolkit" in self._swVersion.lower():
            self.xt_type = "xttoolkit"
        else:
            raise TypeError("UPDATE THIS TEXT.")
```